### PR TITLE
Added leader only response for jsz.

### PR DESCRIPTION
**Pull request by derekcollison**
Speaking with @ripienaar it seemed like a good option to ask the jsz endpoint to only respond if its the leader, since for the metacluster information only the leaders information is useful.

@matthiashanel could you review and possibly add some tests (apologies for that).

Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/monitor.go [hunk 1]
```diff
@@ -2449,12 +2469,18 @@ func (s *Server) HandleJsz(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		return
 	}
+	leader, err := decodeBool(w, r, "leader_only")
```
Comment (matthiashanel) [inline server/monitor.go:2472]: in urls? leader-only

**Review by matthiashanel — APPROVED** [review]
LGTM. name change up to you.

File: server/monitor.go (hunk 1 above)
Comment (derekcollison) [reply server/monitor.go:2472]: sure, or just leader=yes|1?

Comment (matthiashanel) [reply server/monitor.go:2472]: yes just leader should be fine.

**Review by matthiashanel — APPROVED** [review]
LGTM

File: server/monitor.go (hunk 1 above)
Comment (derekcollison) [reply server/monitor.go]: Went with leader-only

File: server/monitor.go [hunk 2]
```diff
@@ -2337,6 +2338,25 @@ func (s *Server) Jsz(opts *JSzOptions) (*JSInfo, error) {
 	if opts.Streams {
 		opts.Accounts = true
 	}
+
+	// Check if we want a response from the leader only.
+	if opts.LeaderOnly {
+		js, cc := s.getJetStreamCluster()
+		if js == nil {
+			// Ignore
+			return nil, ErrJetStreamNotEnabled
```
Comment (ripienaar) [inline server/monitor.go:2347]: Here and on line 2355 if we return Err the error is sent back to the clients, meaning the !leaders still respond but in error.

These 2 lines should be `return nil, nil`

Comment (ripienaar) [reply server/monitor.go:2347]: Ugh, not even ... will dig deeper

Comment (ripienaar) [reply server/monitor.go:2347]: https://github.com/nats-io/nats-server/pull/1902

**Pull request merged**